src/ucp/core/ucp_rkey.c

```diff
@@ -79,6 +102,15 @@ ucs_status_t ucp_ep_rkey_unpack(ucp_ep_h ep, void *rkey_buffer, ucp_rkey_h *rkey
 
     /* Read remote PD map */
     pd_map   = *(uint64_t*)p;
+
+    ucs_trace("unpacking rkey with pd_map 0x%"PRIx64, pd_map);
+
+    if (pd_map == 0) {
+        /* Dummy key return ok */
+        *rkey_p = &dummy_rkey;
```

Comment:
Do you mean to avoid the situation when user has packed and unpacked keys pointing to the same address (which potentially may cause some confusion in user's code)? Is there some other reason to have separate buffer for every packed dummy key?
I'd declare global uint64_t pd_map then to be a dummy rkey buffer.